### DISCUSSION TITLE
Software for opening FlowJo version 9 XML workspaces?

### DISCUSSION THREAD

**Discussion by agartland** [whitews/FlowKit#140] [Q&A]
Hi,
FlowKit seems great, thanks! I have a set of XML FlowJo workspaces from version  9.9.4. I can totally understand why you don't want to spend time covering old versions, but just wondering if you know much about what makes version 9 XML workspaces different from version 10 and if you know of any python packages that can parse them?
Thanks,
Andrew

**Comment by whitews** [comment]
Hi Andrew,

I'm not aware of any other Python packages that can parse FlowJo workspace files. FlowJo 9 uses `.jo` files as its main file format and those files are binary so much more difficult to parse. I know it can export to some XML format as well, but I'm not sure of the differences versus the current v10 format. 

We discussed the issue of legacy v9 .jo files with the FlowJo developers a while back and this led to the addition of a v9 to v10 conversion feature that is now available in v10.8. I believe there is a bulk converter plugin as well. I do know there are a couple issues with converting some files though. You can find info about converting files on BD's site here: 

https://docs.flowjo.com/flowjo/workspaces-and-samples/jo-files-in-v10/

I've converted quite a few .jo files to v10 workspace files and then used FlowKit to parse them successfully. Hope this is helpful,
-Scott
